clientupdate/distsign/distsign.go

```diff
@@ -56,49 +55,90 @@ import (
 )
 
 const (
-	pemTypePrivate = "PRIVATE KEY"
-	pemTypePublic  = "PUBLIC KEY"
+	pemTypeRootPrivate    = "ROOT PRIVATE KEY"
+	pemTypeRootPublic     = "ROOT PUBLIC KEY"
+	pemTypeSigningPrivate = "SIGNING PRIVATE KEY"
+	pemTypeSigningPublic  = "SIGNING PUBLIC KEY"
 
 	downloadSizeLimit    = 1 << 29 // 512MB
 	signingKeysSizeLimit = 1 << 20 // 1MB
 	signatureSizeLimit   = ed25519.SignatureSize
 )
 
-// GenerateKey generates a new key pair and encodes it as PEM.
-func GenerateKey() (priv, pub []byte, err error) {
+// RootKey is a root key used to sign signing keys.
+type RootKey struct {
+	k ed25519.PrivateKey
+}
+
+// GenerateRootKey generates a new root key pair and encodes it as PEM.
+func GenerateRootKey() (priv, pub []byte, err error) {
 	pub, priv, err = ed25519.GenerateKey(rand.Reader)
 	if err != nil {
 		return nil, nil, err
 	}
 	return pem.EncodeToMemory(&pem.Block{
-			Type:  pemTypePrivate,
+			Type:  pemTypeRootPrivate,
 			Bytes: []byte(priv),
 		}), pem.EncodeToMemory(&pem.Block{
-			Type:  pemTypePublic,
+			Type:  pemTypeRootPublic,
 			Bytes: []byte(pub),
 		}), nil
 }
 
-// RootKey is a root key Signer used to sign signing keys.
-type RootKey Signer
+// ParseRootKey parses the PEM-encoded private root key. The key is expected to
+// be in the same format as returned by GenerateRootKey.
+func ParseRootKey(privKey []byte) (*RootKey, error) {
+	k, err := parsePrivateKey(privKey, pemTypeRootPrivate)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse root key: %w", err)
+	}
+	return &RootKey{k: k}, nil
+}
 
 // SignSigningKeys signs the bundle of public signing keys. The bundle must be
```

Comment:
Should we enforce that, by actually calling `ParseSigningKey` on it? (Can you tell I'm getting increasingly paranoid)